### PR TITLE
docs: move Programmer's manual creation to INSTALL.md

### PR DESCRIPTION
**Pull request by neteler**
- move Programmer's manual creation from `doc/development/README.md` to `INSTALL.md`.
- update URLs and wording
- list guides in `doc/development/README.md`

### CODE REVIEW

File: doc/development/README.md [hunk 1]
```diff
@@ -1,43 +1,10 @@
...
-make pdfdocs
-```
+Find below the development and maintenance documentation.
+The API documentation is available in the appropriate places,
+but here is the information relevant to contributing to and
+maintaining GRASS GIS.
+
+- [GRASS Programming Style Guide](./doc/development/style_guide.md)
+- [Guide to contributing on GitHub](./doc/development/github_guide.md)
+- [GRASS Programmer's manual](https://grass.osgeo.org/programming8/)
```
Comment (wenzeslaus) [inline doc/development/README.md:10]: Do the links work for you, e.g., in rich text diff? I think they need to be actually relative, not from the root.

Comment (wenzeslaus) [inline doc/development/README.md:10]: Are you thinking about some "very core" maintenance here when linking only C API doc and not [Python API doc](https://grass.osgeo.org/grass-stable/manuals/libpython/)?

File: INSTALL.md [hunk 2]
```diff
@@ -321,9 +321,42 @@ developers mailing list. See <https://grass.osgeo.org/development/>
...
+the file `lib/html/index.html` which itself refers to further
+document repositories in
+
+```text
+lib/vector/html/index.html
+lib/db/html/index.html
+lib/gis/html/index.html
+```
+
+The master file is: `./grasslib.dox` where all sub-documents have to
```
Comment (wenzeslaus) [inline INSTALL.md:352]: The main file? ...after all the branch migrations.

File: INSTALL.md [hunk 3]
```diff
@@ -321,9 +321,42 @@ developers mailing list. See <https://grass.osgeo.org/development/>
...
+
+This process takes some time. The result will be found in
+the file `lib/html/index.html` which itself refers to further
+document repositories in
+
+```text
+lib/vector/html/index.html
+lib/db/html/index.html
+lib/gis/html/index.html
+```
```
Comment (wenzeslaus) [inline INSTALL.md:350]: How far do you want to go with this list? I suggest just removing it.

File: doc/development/README.md (hunk 1 above)
Comment (neteler) [reply doc/development/README.md:10]: Right, I'll fix this.

**Review by petrasovaa — APPROVED** [review]

File: doc/development/README.md (hunk 1 above)
Comment (wenzeslaus) [reply doc/development/README.md]: Hm, in GitHub under development, I guess devel. I don't really like the name, so perhaps that's why I'm ignoring it.

Comment (neteler) [reply doc/development/README.md]: So it is.

**Review by wenzeslaus — APPROVED** [review]

**Pull request merged**